### PR TITLE
[1LP][RFR] Test to create service dialog using Tower Workflow

### PR DESCRIPTION
**Pull request by nachandr**
## Purpose or Intent
1.Updated AnsibleTowerJobTemplatesCollection so that it can be filtered on Ansible job type.
2.Updated TowerExplorerJobTemplateDetailsView to cover workflow template details page as well.
3.Added test to create service dialog using Tower Workflow

### PRT Run
{{py.test: cfme/tests/infrastructure/test_config_management.py -k "dialog"}}

### PRT Results:
All tests except for the test_config_system_tag[satellite_62]  test passed on both 510 and 511. The failure is not related to the code changes made through this PR.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit eeec6da17cf6872ff3bfba02e08b348032bd0f4e:

`cfme/tests/infrastructure/test_config_management.py`:
- :bangbang: Line 163:101: **E501** *line too long (112 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/ansible_tower/explorer.py [hunk 1]
```diff
@@ -189,7 +192,10 @@ class AnsibleTowerJobTemplatesCollection(BaseCollection, TaggableCollection):
     def all(self):
         """Return entities for all items in Ansible Job templates collection"""
         view = navigate_to(self, "All")
-        return [self.instantiate(e) for e in view.entities.all_entity_names]
+        job_type = self.filters.get('job_type')
+        all_jobs = [self.instantiate(name=e.name,
+            job_type=e.data["type"]) for e in view.entities.get_all()]
```
Comment (digitronik) [inline cfme/ansible_tower/explorer.py]: optional: I would format it like
```
        all_jobs = [
            self.instantiate(name=e.name, job_type=e.data["type"]) for e in view.entities.get_all()
        ]
```

File: cfme/tests/infrastructure/test_config_management.py [hunk 2]
```diff
@@ -160,7 +160,11 @@ def test_ansible_tower_job_templates_tag(request, config_manager, tag):
 
 @pytest.mark.tier(3)
 @pytest.mark.uncollectif(lambda config_manager_obj: config_manager_obj.type != "Ansible Tower")
-def test_ansible_tower_service_dialog_creation_from_template(request, config_manager, appliance):
+@pytest.mark.parametrize('template_type', ['Job Template (Ansible Tower)',
+    'Workflow Template (Ansible Tower)'], ids=['job', 'workflow'],
+    scope='module')
```
Comment (digitronik) [inline cfme/tests/infrastructure/test_config_management.py]: Q: why scope as `module`?

also optional... better to put key value pair for type and name so we can use in other test parametrization like
```
TEMPLATE_TYPE = {
    "job": "Job Template (Ansible Tower)",
    "workflow": "Workflow Template (Ansible Tower)",
}
@pytest.mark.parametrize("template_type", TEMPLATE_TYPE.values(), ids=[TEMPLATE_TYPE.keys()])
def test_foo(template_type):
    pass
```

File: cfme/ansible_tower/explorer.py (hunk 1 above)
Comment (nachandr) [reply cfme/ansible_tower/explorer.py]: Done

File: cfme/tests/infrastructure/test_config_management.py (hunk 2 above)
Comment (nachandr) [reply cfme/tests/infrastructure/test_config_management.py]: Done

**Comment by mshriver** [comment]
@john-dupuy sorry for the rebase you're about to have to do.

**Pull request merged**